ergo-lib/src/wallet/tx_builder.rs

```diff
@@ -131,15 +162,26 @@ impl<S: ErgoBoxAssets + ErgoBoxId + Clone> TxBuilder<S> {
             .iter()
             .filter(|b| b.value >= self.min_change_value)
             .map(|b| {
-                ErgoBoxCandidateBuilder::new(
+                let mut candidate = ErgoBoxCandidateBuilder::new(
                     b.value,
                     change_address_ergo_tree.clone(),
                     self.current_height,
-                )
-                .build()
+                );
+                for token in &b.tokens() {
+                    candidate.add_token(token.clone());
+                }
+                candidate.build()
             })
             .collect();
         output_candidates.append(&mut change_boxes?);
+
+        // Ergo transactions need at least one output
+        // but that single output could just be the change (ex: if you want to send all ERG to a single address)
+        if output_candidates.is_empty() {
```

Comment:
Note: this fixes the issue where you couldn't use the TxBuilder to create a transaction with only change addresses (no explicit outputs). We use this kind of transaction in Yoroi for send-all transactions